src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,93 @@
 package mate.academy.lib;
 
+import java.io.File;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
 public class Injector {
+    private static final String IMPL_PACKAGE = "mate.academy.service.impl";
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (instances.containsKey(interfaceClazz)) {
+            return instances.get(interfaceClazz);
+        }
+        return createInstance(interfaceClazz);
+    }
+
+    private Object createInstance(Class<?> clazz) {
+        try {
+            if (clazz.isInterface()) {
+                clazz = findImplementation(clazz);
+            }
+
+            if (!clazz.isAnnotationPresent(Component.class)) {
+                throw new RuntimeException("Class " + clazz.getName()
+                        + " is missing @Component annotation");
+            }
+
+            Constructor<?> constructor = clazz.getDeclaredConstructor();
+            Object instance = constructor.newInstance();
```

Comment:
The code assumes that the class has a no-argument constructor. If any class does not have a default constructor, this will throw an exception. Consider handling this case or documenting the requirement for a no-argument constructor.